src/main/java/net/dries007/tfc/api/calendar/ICalendarFormatted.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Work under Copyright. Licensed under the EUPL.
+ * See the project README.md and LICENSE.txt for more information.
+ */
+
+package net.dries007.tfc.api.calendar;
+
+import net.minecraft.world.server.ServerWorld;
+
+public interface ICalendarFormatted extends ICalendar
+{
+    /* Total calculation methods */
+
+    static long getTotalMonths(long time, long daysInMonth)
+    {
+        return time / (daysInMonth * TICKS_IN_DAY);
+    }
+
+    static long getTotalYears(long time, long daysInMonth)
+    {
+        return 1000 + (time / (12 * daysInMonth * TICKS_IN_DAY));
+    }
+
+    /* Fraction Calculation Methods */
+
+    static int getMinuteOfHour(long time)
+    {
+        return (int) ((time % TICKS_IN_HOUR) / TICKS_IN_MINUTE);
+    }
+
+    static int getHourOfDay(long time)
+    {
+        return (int) ((time / TICKS_IN_HOUR) % HOURS_IN_DAY);
+    }
+
+    static int getDayOfMonth(long time, long daysInMonth)
+    {
+        return 1 + (int) ((time / TICKS_IN_DAY) % daysInMonth);
+    }
+
+    /* Format Methods */
+
+    /* todo
+    static ITextComponent getTimeAndDate(long time, long daysInMonth)
+    {
+        return getTimeAndDate(getHourOfDay(time), getMinuteOfHour(time), getMonthOfYear(time, daysInMonth), getDayOfMonth(time, daysInMonth), getTotalYears(time, daysInMonth));
+    }
+    */
+
+    /* todo
+    static ITextComponent getTimeAndDate(int hour, int minute, Month month, int day, long years)
+    {
+        String monthName = TerraFirmaCraft.getProxy().getMonthName(month, false);
+        return TerraFirmaCraft.getProxy().getDate(hour, minute, monthName, day, years);
+    }*/
+
+    static Month getMonthOfYear(long time, long daysInMonth)
+    {
+        return Month.valueOf((int) ((time / (TICKS_IN_DAY * daysInMonth)) % 12));
+    }
+
+    /**
+     * Gets the current time, according to the specific calendar's offset and settings
+     *
+     * @return a time
+     */
+    @Override
+    long getTicks();
+
+    /**
+     * Gets the number of days in a month, from the calendar's instance values
+     *
+     * @return a number of days in a month
+     */
+    long getDaysInMonth();
+
+    /* todo
+    default ITextComponent getTimeAndDate()
```

Comment:
If this is left out due to proxy not existing, it's very much the same as 1.12, just using `DistExecutor`. Although we should upgrade our forge version to get the safer classloaded changes before implenting.
```java
// in TerraFirmaCraft:
private static final IProxy PROXY = DistExecutor.safeRunForDist(() -> () -> new ClientProxy(), () -> () -> new ServerProxy());
```
The lambdas are important, do not convert to method references (suppress warnings if nessecary).

Reply:
Not only because of the proxy changes: We now should re-evaluate if this is needed on the server side, and if it is, implement using `ITextComponent` instead of `String`.